### ISSUE TITLE
Fix Request for AvoidUnnecessaryStringBuilderCreation: false positive for non-append operations 

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#427]
Method reverse is used on the buffer which cannot be replaced by a '+'. This validates the creation of a StringBuilder
Example:
```
StringBuilder timestampBuilder = new StringBuilder(new SimpleDateFormat(TIME_STAMP_FORMAT).format(new Date()));
String reversedTimeStamp = timestampBuilder.reverse().append(userId).toString(); // <--- false positive
```

**Comment by jborgers** [comment]
There is more wrong, local var in stead of return,..Created extra unit tests.
